### PR TITLE
Fix bicleaner download issue in CI

### LINKED ISSUE #429: Bicleaner AI task fails to fetch multilingual model
https://firefox-ci-tc.services.mozilla.com/tasks/PC5bhFKXQ_6b3ZaD5PRklw/runs/0/logs/public/logs/live.log

```
Downloading a model for en-ca
Failed. Trying ca-en
Traceback (most recent call last):
  File "/usr/lib/python3.10/shutil.py", line 816, in move
    os.rename(src, real_dst)
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/ca-en' -> '/tmp/bicleaner-ai-en-ca'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/builds/worker/checkouts/vcs/pipeline/bicleaner/download_pack.py", line 123, in <module>
    main()
  File "/builds/worker/checkouts/vcs/pipeline/bicleaner/download_pack.py", line 114, in main
    download(
  File "/builds/worker/checkouts/vcs/pipeline/bicleaner/download_pack.py", line 82, in download
    shutil.move(pack_path, new_name)
  File "/usr/lib/python3.10/shutil.py", line 836, in move
    copy_function(src, real_dst)
  File "/usr/lib/python3.10/shutil.py", line 434, in copy2
    copyfile(src, dst, follow_symlinks=follow_symlinks)
  File "/usr/lib/python3.10/shutil.py", line 254, in copyfile
    with open(src, 'rb') as fsrc:
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/ca-en'
```

Blocks #384

### PR DESCRIPTION
**Pull request by gregtatum**
Resolves #429

I also added a structured logger, which I'm finding helpful in taskcluster logs.

### CODE REVIEW

**Comment by eu9ene** [comment]
I ran into this for lt-en too ([log](https://firefox-ci-tc.services.mozilla.com/tasks/ObT3cJvdScG5omw-B4VtDw/runs/0/logs/public/logs/live.log)). I wonder what's the difference with en-ru/ru-en runs where it succeeds. I see that bicleaner-ai has a [model](https://huggingface.co/bitextor/bicleaner-ai-full-en-lt) for en-lt that it seems is successfully downloaded.

**Comment by gregtatum** [comment]
I changed the implementation to just look for the existence of the downloads, rather than status codes reported by bicleaner. It seems to work.

Here is an example of what the logs look like with the structured logger.

```
[task 2024-02-09T17:43:05.687Z] ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
[task 2024-02-09T17:43:05.687Z] mtdata 0.4.0 requires requests==2.26.0, but you have requests 2.28.2 which is incompatible.
[task 2024-02-09T17:43:05.687Z] Successfully installed absl-py-1.4.0 astunparse-1.6.3 bicleaner-ai-2.0.0 bicleaner-ai-glove-0.2.1 bicleaner-hardrules-2.7.0 cachetools-5.3.0 certifi-2022.12.7 charset-normalizer-3.1.0 click-8.1.3 exceptiongroup-1.1.1 fastspell-0.5 fasttext-0.9.2 filelock-3.12.0 flatbuffers-23.3.3 fuzzywuzzy-0.18.0 gast-0.4.0 google-auth-2.17.3 google-auth-oauthlib-0.4.6 google-pasta-0.2.0 grpcio-1.54.0 h5py-3.8.0 huggingface-hub-0.11.1 idna-3.4 iniconfig-2.0.0 joblib-1.2.0 keras-2.11.0 levenshtein-0.20.9 libclang-16.0.0 markdown-3.4.3 markupsafe-2.1.2 numpy-1.24.2 oauthlib-3.2.2 opt-einsum-3.3.0 packaging-23.1 pluggy-1.0.0 protobuf-3.19.6 psutil-5.9.5 pyasn1-0.4.8 pyasn1-modules-0.2.8 pybind11-2.11.1 pytest-7.3.1 python-levenshtein-0.20.9 pyyaml-6.0 rapidfuzz-2.15.1 regex-2023.3.23 requests-2.28.2 requests-oauthlib-1.3.1 rsa-4.9 sacremoses-0.0.53 scikit-learn-1.2.2 scipy-1.10.1 sentencepiece-0.1.98 tensorboard-2.11.2 tensorboard-data-server-0.6.1 tensorboard-plugin-wit-1.8.1 tensorflow-2.11.1 tensorflow-estimator-2.11.0 tensorflow-io-gcs-filesystem-0.32.0 termcolor-2.2.0 threadpoolctl-3.1.0 tokenizers-0.13.3 tomli-2.0.1 toolwrapper-2.1.0 tqdm-4.65.0 transformers-4.26.0 typing-extensions-4.5.0 urllib3-1.26.15 werkzeug-2.2.3 wheel-0.40.0 wrapt-1.15.0
[task 2024-02-09T17:43:06.417Z] [download_pack] Attempt 1 of 3: Downloading a model for ru-en
[task 2024-02-09T17:43:07.316Z] [download_pack] Attempt 2 of 3. Downloading a model for en-ru
[task 2024-02-09T17:43:08.229Z] [download_pack] Attempt 3 of 3. Downloading the multilingual model en-xx
[task 2024-02-09T17:43:27.090Z] [download_pack] Compress the downloaded pack.
[task 2024-02-09T17:43:27.090Z] [download_pack] pack_path: "/tmp/bicleaner/en-xx"
[task 2024-02-09T17:43:27.091Z] [download_pack] new_name: "/tmp/bicleaner/bicleaner-ai-ru-en"
[task 2024-02-09T17:43:27.091Z] [download_pack] Compressing /tmp/bicleaner/bicleaner-ai-ru-en
```

**Comment by gregtatum** [comment]
Marking as draft until I handle logging feedback from #420.

**Comment by gregtatum** [comment]
I removed the structured logging, so this is ready for review.

File: pipeline/bicleaner/download_pack.py [hunk 1]
```diff
@@ -50,39 +50,68 @@ def _compress_dir(dir_path: str, compression_cmd: str) -> str:
...
+    if result.returncode != 0:
+        print(result.stdout)
+        print(result.stderr)
+        result.check_returncode()
+
+
 def download(src: str, trg: str, output_path: str, compression_cmd: str) -> None:
-    tmp_dir = tempfile.gettempdir()
+    tmp_dir = os.path.join(tempfile.gettempdir(), "bicleaner")
+    if not os.path.exists(tmp_dir):
```
Comment (bhearsum) [inline pipeline/bicleaner/download_pack.py:63]: Was not creating the temp directory the problem?

**Review by bhearsum — APPROVED** [review]

File: pipeline/bicleaner/download_pack.py (hunk 1 above)
Comment (gregtatum) [reply pipeline/bicleaner/download_pack.py:63]: This just helped in debugging the issue. Locally this put it in the temp directory with everything else that was in the temp directory, so it was hard to see what was being saved there.

**Review by eu9ene — APPROVED** [review]

**Pull request merged**